format.go

```diff
@@ -342,23 +351,60 @@ func (a *unknownAny) ProtoMessage() {
+type FormatOptions struct {
+	// EmitJSONDefaultFields flag, when true, includes empty/default values in the output.
+	// FormatJSON only flag.
+	EmitJSONDefaultFields bool
+
+	// AllowUnknownFields is an option for the parser. When true,
+	// it accepts input which includes unknown fields. These unknown fields
+	// are skipped instead of returning an error.
+	// FormatJSON only flag.
+	AllowUnknownFields bool
+
+	// IncludeTextSeparator is true then, when invoked to format multiple messages,
+	// all messages after the first one will be prefixed with the
+	// ASCII 'Record Separator' character (0x1E).
+	// It might be useful when the output is piped to another grpcurl process.
+	// FormatText only flag.
+	IncludeTextSeparator bool
+}
+
+// RequestParserAndFormatter returns a request parser and formatter for the
 // given format. The given descriptor source may be used for parsing message
-// data (if needed by the format). The flags emitJSONDefaultFields and
-// includeTextSeparator are options for JSON and protobuf text formats,
-// respectively. Requests will be parsed from the given in.
-func RequestParserAndFormatterFor(format Format, descSource DescriptorSource, emitJSONDefaultFields, includeTextSeparator bool, in io.Reader) (RequestParser, Formatter, error) {
+// data (if needed by the format).
+// It accepts a set of options. The field EmitJSONDefaultFields and IncludeTextSeparator
+// are options for JSON and protobuf text formats, respectively. The AllowUnknownFields field
+// is a JSON-only format flag.
+// Requests will be parsed from the given in.
+func RequestParserAndFormatter(format Format, descSource DescriptorSource, in io.Reader, opts FormatOptions) (RequestParser, Formatter, error) {
 	switch format {
 	case FormatJSON:
 		resolver := AnyResolverFromDescriptorSource(descSource)
-		return NewJSONRequestParser(in, resolver), NewJSONFormatter(emitJSONDefaultFields, anyResolverWithFallback{AnyResolver: resolver}), nil
+		unmarshaler := jsonpb.Unmarshaler{AnyResolver: resolver, AllowUnknownFields: opts.AllowUnknownFields}
+		return NewJSONRequestParserWithUnmarshaler(in, unmarshaler), NewJSONFormatter(opts.EmitJSONDefaultFields, anyResolverWithFallback{AnyResolver: resolver}), nil
 	case FormatText:
-		return NewTextRequestParser(in), NewTextFormatter(includeTextSeparator), nil
+		return NewTextRequestParser(in), NewTextFormatter(opts.IncludeTextSeparator), nil
 	default:
 		return nil, nil, fmt.Errorf("unknown format: %s", format)
 	}
 }
 
+// RequestParserAndFormatterFor returns a request parser and formatter for the
+// given format. The given descriptor source may be used for parsing message
+// data (if needed by the format). The flags emitJSONDefaultFields and
+// includeTextSeparator are options for JSON and protobuf text formats,
+// respectively. Requests will be parsed from the given in.
+// This function is deprecated. Please use RequestParserAndFormatter instead.
+// DEPRECATED
```

Comment:
I am not sure what's project's convention to mark functions as deprecated. I explained it in the comment and added `// DEPRECATED` line. The latter works nicely with editors, for example GoLand, that inform when a deprecated function is used.

Reply:
https://github.com/golang/go/wiki/Deprecated

This is fine. I may tweak it in a later commit. More idiomatic for Go would be to take the last two lines here and change them to something like so:

```go
// Deprecated: Use RequestParserAndFormatter instead.
```